src/internal/packager/images/pull.go

```diff
@@ -188,7 +188,7 @@ func (i *ImageConfig) PullAll() ([]ImgInfo, error) {
 	var progressBarWaitGroup sync.WaitGroup
 	progressBarWaitGroup.Add(1)
 	updateText := fmt.Sprintf("Pulling %d images", imageCount)
-	go utils.RenderProgressBarForLocalDirWrite(i.ImagesPath, totalBytes, &progressBarWaitGroup, doneSaving, updateText, updateText)
+	go utils.RenderProgressBarForLocalDirWrite(i.ImagesPath, totalBytes, &progressBarWaitGroup, doneSaving, nil, updateText, updateText)
```

Comment:
There's some other logic happening with the wait group, see [onLayerWritingError](https://github.com/defenseunicorns/zarf/blob/6a6be5439eec1a1740ab7b8626ae79a0863878e8/src/internal/packager/images/pull.go#L325)

Reply:
onLayerWritingError should be refactored to use an `errorChan` instead of `doneSaving`